### ISSUE TITLE
Does the -destroy respect "protected" status?

### ISSUE THREAD

**Issue by jamesdehart** [Praqma/helmsman#621]
Hey all,

I hope this is the right place to post this. 

The [docs](https://github.com/Praqma/helmsman/blob/master/docs/how_to/misc/protect_namespaces_and_releases.md) says

## When a release (application) is protected, it CANNOT:
* deleted
* upgraded
* moved to another namespace

When I was using the `-destroy` switch it destoryed all the apps even the protected apps. I don't have any real experence with go but I think [this](https://github.com/Praqma/helmsman/blob/master/internal/app/decision_maker.go#L97-L103) code is not checking if the chart is protected. 

I love using the `-destroy` flag when cleaning up the env for the next deployment. I was not sure if I was reading the docs wrong. 

Please advise.

**Comment by luisdavim** [comment]
Hi, I actually don't know if this is on purpose or an oversight but it seems like the protection only applies to the `apply` actions.
Since it's better to be safe than sorry, I think it's better to honour the protection flag on destroy as well, I'll open a PR for this.

**Comment by luisdavim** [comment]
@jamesdehart can you try a build from https://github.com/Praqma/helmsman/pull/622 ?

**Comment by jamesdehart** [comment]
Sure let me give it a try. Thanks!

**Comment by jamesdehart** [comment]
@luisdavim Should I compile your branch or are there compiled bits somewhere?

**Comment by luisdavim** [comment]
I don't think we have builds for PRs so you'll have to clone, checkout the branch and run `make build`
